src/payment/strategies/braintree/braintree-paypal-payment-strategy.ts

```diff
@@ -28,29 +29,26 @@ export default class BraintreePaypalPaymentStrategy implements PaymentStrategy {
         private _credit: boolean = false
     ) {}
 
-    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
         const { braintree: braintreeOptions, methodId } = options;
 
-        this._paymentMethod = this._store.getState().paymentMethods.getPaymentMethod(methodId);
-
-        if (this._paymentMethod && this._paymentMethod.nonce) {
-            return Promise.resolve(this._store.getState());
+        if (!this._paymentMethod || !this._paymentMethod.nonce) {
+            this._paymentMethod = this._store.getState().paymentMethods.getPaymentMethodOrThrow(methodId);
         }
 
-        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
-            .then(state => {
-                this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+        if (this._paymentMethod.clientToken) {
+            return this._loadPaypal(braintreeOptions);
+        } else {
```

Comment:
minor, if you are returning inside `if`, you don't need `else`. It just makes it easier to read/follow